docs/hub/information_extraction.md

```diff
@@ -0,0 +1,103 @@
+# Customer Information Extraction
+
+In this guide, we'll walk through how to extract customer lead information using OpenAI's API and Pydantic. This use case is essential for seamlessly automating the process of extracting specific information from a context.
+
+If you want to try this out via `instructor hub`, you can pull it by running:
+
+```bash
+instructor hub pull --slug information_extraction --py > information_extraction.py
+```
+
+## Motivation
+
+You could potentially integrate this into a chatbot to extract relevant user information from user messages. With the use of machine learning driven validation it would reduce the need for a human to verify the information.
+
+## Defining the Structure
+
+We'll model a customer lead as a Lead object, including fields for the name and phone number. We'll use a Pydantic field_validator for data validation and formatting.
+
+## Extracting Lead Information
+
+To extract lead information, we create the `parse_lead_from_message` function which integrates Instructor. It calls OpenAI's API, processes the text, and returns the extracted lead information as a Lead object.
+
+## Evaluating Lead Extraction
+
+To showcase the `parse_lead_from_message` function we can provide sample user messages that may be obtained from a dialogue with a chatbot assistant.
+
+```python
+import instructor
+from openai import OpenAI
+from pydantic import BaseModel, field_validator
+
+
+class Lead(BaseModel):
+    name: str
+    phone_number: str
+
+    @field_validator("phone_number")
+    @classmethod
+    def validate_phone_number(cls, value):
+        # Remove any non-digit characters from the phone number
+        digits_only = ''.join(filter(str.isdigit, value))
+
+        # Check if the cleaned phone number has exactly 10 digits
+        if len(digits_only) != 10:
+            raise ValueError("Invalid phone number. Please provide exactly 10 digits.")
+
+        # Format the validated phone number as XXX-XXX-XXXX
+        formatted_number = f"{digits_only[:3]}-{digits_only[3:6]}-{digits_only[6:]}"
+
+        return formatted_number
+
+    # Can define some function here to send Lead information to a database using an API
+
+
+client = instructor.patch(OpenAI())
+
+
+def parse_lead_from_message(user_message: str):
+    return client.chat.completions.create(
+        model="gpt-4",
+        response_model=Lead,
+        messages=[
+            {
+                "role": "system",
+                "content": "You are a data extraction system that extracts a user's name and phone number from a message.",
+            },
+            {
+                "role": "user",
+                "content": f"Extract the user's lead information from this user's message: {user_message}",
+            },
+        ],
+    )
+
+
+if __name__ == "__main__":
+    lead = parse_lead_from_message(
+        "Yes, that would be great if someone can reach out my name is Patrick King 9172234587"
```

Comment:
```suggestion
        "Yes, that would be great if someone can reach out my name is Patrick King 9175554587"
```

Reply:
One caveat of using that pydantic PhoneNumber type is that it requires the "+[some country code]" at the beginning of the number or else it comes back as invalid without a useful error message. So I'll use "+19175554587" in the example.